puppet/modules/jenkins_job_builder/manifests/config.pp

```diff
@@ -6,46 +6,28 @@
   String $password,
   Integer[0] $jenkins_jobs_update_timeout = 600,
   String $command_arguments = 'foreman-infra-jenkins-job-update',
+  String $git_project_name = 'jenkins-jobs',
+  String $git_repo = 'https://github.com/theforeman/jenkins-jobs.git',
+  Optional[String] $git_ref = undef,
 ) {
   $config_name = $name
   $directory = '/etc/jenkins_jobs'
   $inifile = "${directory}/jenkins_jobs_${config_name}.ini"
 
-  file { "${directory}/${config_name}":
-    ensure  => directory,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0755',
-    recurse => true,
-    purge   => true,
-    force   => true,
-    source  => "puppet:///modules/jenkins_job_builder/${config_name}",
+  vcsrepo { "${directory}/${git_project_name}",
+    ensure   => latest,
+    provider => git,
+    source   => $git_repo,
+    revision => $git_ref,
     notify  => Exec["jenkins-jobs-update-${config_name}"],
   }
 
-  cron { "jenkins-jobs-update-${config_name}-delete-old":
-    ensure      => absent,
-    command     => "timeout 1h jenkins-jobs --conf ${inifile} update --delete-old ${directory}/${config_name} > /var/cache/jjb.xml",
-    hour        => 0,
-    minute      => 10,
-    environment => 'PATH=/bin:/usr/bin:/usr/sbin',
-    require     => File[$inifile],
-  }
-
   exec { "jenkins-jobs-update-${config_name}":
-    command => "jenkins-jobs --conf ${inifile} update ${directory}/${config_name} ${command_arguments}",
+    command => "jenkins-jobs --conf ${inifile} update . ${command_arguments}",
     timeout => $jenkins_jobs_update_timeout,
     path    => '/bin:/usr/bin:/usr/local/bin',
     require => File[$inifile],
-  }
-
-  cron { "remove-unmanaged-jobs-${config_name}":
-    ensure      => absent,
-    command     => "ruby ${directory}/${config_name}/unmanaged_jobs.rb ${inifile}",
-    hour        => 1,
-    minute      => 10,
-    environment => 'PATH=/bin:/usr/bin:/usr/sbin',
-    require     => File[$inifile],
+    cwd     => "${directory}/${git_project_name}/${config_name}",
```

Comment:
What's the benefit of setting `cwd` rather than passing an absolute path in the command? Absolute paths are easier to understand in debug logs since AFAIK `cwd` isn't really logged clearly.